DEVELOPERS.md

```diff
@@ -0,0 +1,75 @@
+## Developing Icetea
+This document describes some information that is relevant for developers of Icetea.
+
```

Comment:
A general comment: Always have an empty line after a heading.
Are you sure you want to include this doc into the public Mbed OS documentation site? I'm not really getting the point of this doc.

- some information? -> somethin like: This document is targeted for anyone developing the tool further. 

Or is this really targeted to tool developers? If it is, why are you talking about how to test using Icetea?

Reply:
This document should not (?) be included, only the README.md as far as I know?

Reply:
And yes, I'll rephrase this slightly.